src/graphics/mesh3d.rs

```diff
@@ -0,0 +1,1106 @@
+                    | obj::raw::object::Polygon::PN(ref vec)
+                        if vec.len() == 3 =>
+                    {
+                        for &(pi, _) in vec {
+                            map(pi)?
+                        }
+                    }
+                    obj::raw::object::Polygon::PTN(ref vec) if vec.len() == 3 => {
+                        for &(pi, _, _) in vec {
+                            map(pi)?
+                        }
+                    }
+                    _ => {
+                        return Err(obj::ObjError::Load(obj::LoadError::new(
+                            obj::LoadErrorKind::UntriangulatedModel,
+                            "Meshes must be triangulated",
+                        )))
+                    }
+                }
+            }
+        }
+        Ok((verts, inds))
+    }
+}
+
+/// Model is an abstracted type for holding things like obj, gltf, or anything that may be made up of multiple meshes.
+#[derive(Debug, Default)]
+pub struct Model {
+    /// The meshes that make up the model
+    pub meshes: Vec<Mesh3d>,
+}
+
+impl Drawable3d for Model {
+    fn draw(&self, canvas: &mut Canvas3d, param: impl Into<DrawParam3d>) {
+        let param = param.into();
+        let param = if let Transform3d::Values { offset, .. } = param.transform {
+            if offset.is_none() {
+                param.offset(self.to_aabb().unwrap_or_default().center)
+            } else {
+                param
+            }
+        } else {
+            param
+        };
+        for mesh in self.meshes.iter() {
+            canvas.push_draw(Draw3d::Mesh { mesh: mesh.clone() }, param);
+        }
+    }
+}
+
+impl Model {
+    /// Create a model from a mesh
+    pub fn from_mesh(mesh: Mesh3d) -> Self {
+        Model { meshes: vec![mesh] }
+    }
+    /// Load gltf or obj depending on extension type
+    #[cfg(all(feature = "obj", feature = "gltf"))]
+    pub fn from_path(
+        gfx: &mut impl HasMut<GraphicsContext>,
+        path: impl AsRef<Path>,
+        image: impl Into<Option<Image>>,
+    ) -> GameResult<Self> {
+        if let Some(extension) = path.as_ref().extension() {
+            if extension == "obj" {
+                Model::from_obj(gfx, path, image)
+            } else if extension == "gltf" || extension == "glb" {
+                Model::from_gltf(gfx, path)
+            } else {
+                Err(GameError::CustomError("Not a obj or gltf file".to_string()))
+            }
+        } else {
+            Err(GameError::CustomError(
+                "Failed to get extension".to_string(),
+            ))
+        }
+    }
+
+    /// Load obj file. Only triangulated obj's are supported. Keep in mind mtl file's currently don't affect the obj's rendering
+    #[cfg(feature = "obj")]
+    pub fn from_obj(
+        gfx: &mut impl HasMut<GraphicsContext>,
+        path: impl AsRef<Path>,
+        image: impl Into<Option<Image>>,
+    ) -> GameResult<Self> {
```

Comment:
It would be nice to have a `from_obj_bytes` function, like for `Image`, `SoundData`, `Shader`, etc

Same for `from_gltf` and `from_path`